### ISSUE TITLE
Detailed Error Messaging: Non-PDF File

### ISSUE THREAD

**Issue by cholly75** [ustaxcourt/ef-cms#8431]
As a petitioner or practitioner, so that I know how to correct problems when uploading files, I need more detailed responses when I upload a file that is not a PDF.

Currently when a user uploads a file in DAWSON, if the file does not meet certain criteria we display an error in the UI and ask them to try again. Even though there are a few commonly seen specific causes for the error, we do not give the user any indication of what the problem might be to help them successfully upload the file on subsequent retries.

We would like to display custom error messaging in these cases to help our external users understand what problems they need to address in their file before attempting to re-upload it. We have been able to connect several root causes for upload problems with specific logfile messages in Kibana, and we should display helpful information to the user based on the error seen in those messages.

## Acceptance Criteria
* When a petitioner attempts to upload a file, if the uploaded PDF is not a PDF, the UI explains in some way to the user that the PDF they are trying to upload may not in fact an actual PDF file, and asks them to check/ensure the file is a genuine PDF file before trying to upload again.
* When a practitioner attempts to upload a file, if the uploaded PDF is not a PDF, the UI explains in some way to the user that the PDF they are trying to upload may not in fact an actual PDF file, and asks them to check/ensure the file is a genuine PDF file before trying to upload again.
* Occurrences of these errors manifest in the back end Kibana log files for tracking

## Notes
- We will probably want to adjust the wording in the current error message that is at the end of the workflow when this story is implemented.  Once this story is in place, the current error message will no longer be valid:
<img width="697" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMjZPQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--e2fe6530f6b355870b865161c6b7c9da6ab10bae/image.png" alt="image.png" />

- Here is the error message we see in the logs:
![image.png](https://images.zenhubusercontent.com/60e7a3b441280d0c68e72525/66a990aa-6bb2-4379-ac49-2e1fd13268cf)

## Story Definition of Ready (updated on 12/23/22)
The following criteria must be met in order for the user story to be picked up by the Flexion development team.
The user story must: 
- [ ]  Is framed in business/user need, the value has been addressed.
- [ ] Includes acceptance criteria
- [ ] Has been refined
- [ ] Pre conditions have been satisfied. 
**Process:** 
Flexion developers and designers will test if the story meets acceptance criteria and test cases in Flexion dev and staging environments (“standard testing”). If additional acceptance criteria or testing scenarios are discovered while the story is in progress, a new story should be created, added to the backlog and prioritized by the product owner. 

## Definition of Done (Updated 5-19-22)
**Product Owner**
 - [ ]  Acceptance criteria have been met and validated on the Court's migration environment
 - [ ] Add scenario to testing document, if applicable (https://docs.google.com/spreadsheets/d/1FUHKC_YrT-PosaWD5gRVmsDzI1HS_U-8CyMIb-qX9EA/edit?usp=sharing)

**UX**
 - [ ] Business test scenarios have been refined to meet all acceptance criteria 
 - [ ] Usability has been validated
 - [ ] Wiki has been updated (if applicable) 
 - [ ] Story has been tested on a mobile device (for external users only)

**Engineering**
 - [ ] Automated test scripts have been written, including visual tests for newly added PDFs
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [ ] Acceptance criteria for the story has been met
 - [ ] If there are special instructions in order to deploy into the next environment, add them as a comment in the story
 - [ ] Reviewed by UX on a deployed environment.
 - [ ] Reviewed by PO on a deployed environment. Can be deployed to the Court's `test` environment if prod-like data is required. Otherwise deployed to any `experimental` environment.
 - [ ] Deployed to the Court's `staging` environment.

**Comment by ttlenard** [comment]
## Test Cases

#### 1) Petitioner uploads a non-PDF file for their STIN; Petitioner receives error message immediately after upload; Petitioner is unable to proceed to the next step.

- Log in as a Petitioner
- Create a case
- At the step to upload a STIN, Click on the Choose File button
- Choose a file from your computer that is not a PDF.

Expected Results:
- [ ] User receives the red error message banner with the text from the mockup at the top of the screen.
- [ ] User receives error message text in red text underneath the file name.
- [ ] User is unable to proceed to the next screen when they click "Continue to Step 2 of 5" button; Error message persists until user uploads an appropriate PDF file.
- [ ] Upon uploading an appropriate PDF document, user is able to click on the "Continue to Step 2 of 5" button to get to the next step without the error message.


*Repeat this test as a Private Practitioner
- [ ] Test passes as Private Practitioner



#### 2) Kibana logs display the error message that the user experienced when they uploaded the non-PDF file STIN.

- Log into Kibana
- Filter by the User that you were logged in as for Test Case 1

Expected Results:
- [ ] Error message that the user experienced is present in the Kibana logs



#### 3) Petitioner uploads a non-PDF file for their Petition; Petitioner receives error message immediately after upload; Petitioner is unable to proceed to the next step.

- Continuing from Test Case 1
- At the step to upload a Petition, Click on the Choose File button
- Choose a file from your computer that is not a PDF

Expected Results:
- [ ] User receives the red error message banner with the text from the mockup at the top of the screen.
- [ ] User receives error message text in red text underneath the file name.
- [ ] User is unable to proceed to the next screen when they click "Continue to Step 3 of 5" button; Error message persists until user uploads an appropriate PDF file.
- [ ] Upon uploading an appropriate PDF document, user is able to click on the "Continue to Step 3 of 5" button to get to the next step without the error message.


*Repeat this test as a Private Practitioner
- [ ] Test passes as Private Practitioner


#### 4) Kibana logs display the error message that the user experienced when they uploaded the non-PDF file Petition.

- Log into Kibana
- Filter by the User that you were logged in as for Test Case 1 & 3

Expected Results:
- [ ] Error message that the user experienced is present in the Kibana logs



#### 5) Petitioner uploads a non-PDF file for their CDS; Petitioner receives error message immediately after upload; Petitioner is unable to proceed to the next step.

- Continuing from Test Case 3
- Be sure that you select that you are filing the Petition on behalf of a Business
- Select Corporation (You'll need to repeat this test by selecting all of the business types)
- Fill out the business information
- Click on the Choose File button
- Choose a file from your computer that is not a PDF.

Expected Results:
- [ ] User receives the red error message banner with the text from the mockup at the top of the screen.
- [ ] User receives error message text in red text underneath the file name.
- [ ] User is unable to proceed to the next screen when they click "Continue to Step 4 of 5" button; Error message persists until user uploads an appropriate PDF.
- [ ] Upon uploading an appropriate PDF document, user is able to click on the "Continue to Step 4 of 5" button to get to the next step without the error message.
- [ ] User is able to complete the Create a Case workflow and submit their Petition Successfully.


*Repeat this test by selecting all of the business types available
*Repeat this test as a Private Practitioner
- [ ] Test passes as Private Practitioner



#### 6) Kibana logs display the error message that the user experienced when they uploaded the non-PDF file CDS.

- Log into Kibana
- Filter by the User that you were logged in as for Test Case 1, 3, and 5

Expected Results:
- [ ] Error message that the user experienced is present in the Kibana logs



#### 7) Petitioner files a document on their case; uploads a non-PDF File; Petitioner receives error message immediately after upload; Petitioner is unable to proceed to the next step.

- Log in as a Petitioner
- Click on the docket number of one of your existing cases
- Click on the File a Document button
- Click on the Ok, I'm Ready to File button
- Select a document type, click continue
- Click on the Choose File button
- Choose a file from your computer that is not a PDF.

Expected Results:
- [ ] User receives the red error message banner with the text from the mockup at the top of the screen.
- [ ] User receives error message text in red text underneath the file name.
- [ ] User is unable to proceed to the next screen when they click the Review Filing button; Error message persists until user uploads an appropraite PDF file.
- [ ] User is able to complete the File a document workflow and Submit their filing successfully


*Repeat this test with a Private Practitioner and an IRS Practitioner.
- [ ] Test passes as Private Practitioner
- [ ] Test Passes as an IRS Practitioner



#### 8) Kibana logs display the error message that the user experienced when they uploaded the non-PDF File document.

- Log into Kibana
- Filter by the User that you were logged in as for Test Case 7

Expected Results:
- [ ] Error message that the user experienced is present in the Kibana logs


#### 9) Petitioner files a supporting document with their filing on their case; uploads a non-PDF file for the supporting document; Petitioner receives error message immediately after upload; Petitioner is unable to proceed to the next step.

- Log in as a Petitioner
- Click on the docket number of one of your existing cases
- Click on the File a Document button
- Click on the Ok, I'm Ready to File button
- Select a document type, click continue
- Click on the Choose File button
- Upload an appropriate PDF file
- Click on Add supporting document
- Select a supporting document type from the dropdown
- Click on the Choose File button
- Choose a file from your computer that is not a PDF.

Expected Results:
- [ ] User receives the red error message banner with the text from the mockup at the top of the screen.
- [ ] User receives error message text in red text underneath the file name.
- [ ] User is unable to proceed to the next screen when they click the Review Filing button; Error message persists until user uploads an appropriate PDF file.
- [ ] User is able to complete the File a document workflow and Submit their filing successfully

*Repeat this test by adding multiple supporting documents, ensuring that for each supporting document, you receive the error message (can add up to 5 supporting documents for each filing)
*Repeat this test with a Private Practitioner and an IRS Practitioner.
- [ ] Test passes as Private Practitioner
- [ ] Test Passes as an IRS Practitioner



#### 10) Kibana logs display the error message that the user experienced when they uploaded the non-PDF File supporting document.

- Log into Kibana
- Filter by the User that you were logged in as for Test Case 9

Expected Results:
- [ ] Error message that the user experienced is present in the Kibana logs



#### 11) Repeat test cases 1-10 using a mobile device; error messages received on Mobile device screens match mockups.

- [ ] Test cases 1-10 pass on mobile device

**Comment by jhansen-flexion** [comment]
Blocked - see 10001
